### DISCUSSION TITLE
Add Spanish Model - How to Help

### DISCUSSION THREAD

**Discussion by aeschylus** [winkjs/wink-nlp#120] [New Features & Ideas]
I run a language learning software company. We would benefit greatly from quality in-browser (and nodejs) nlp tools. How can we get started adding support for Spanish, French, German, and other languages?

**Comment by sanjayaksaxena** [comment]
Hello @aeschylus 

We are delighted to know that you found winkNLP useful.

Currently our model building solution is not open source as refinements/improvements/development along with documentation need work. We have recently initiated work on the same. It is going to take time to complete as open source always competes with our day jobs! We will keep you posted on the progress. We are working on model for French as of now.

Would like to know what is the minimal capability you need apart apart from tokenization; currently English model supports sentence boundary detection (sbd), negation handling, sentiment analysis, part-of-speech (pos) tagging, named entity recognition (ner), and custom entities recognition (cer).

Do you think you can contribute to the development of the French language model?

Best,
Sanjaya

**Reply by aeschylus** [reply]
Yes, I'd love to contribute to a french model if there's a clear way for me to do so. We're mostly interested in part-of-speech tagging and entity recognition (distant second). In particular, it would be very helpful if there were a way to get all valid conjugations for a verb, and get the infinitive from a given conjugation of a verb. We have a slow, expensive, and kludgy way of doing this currently, and it would be nice to have the plumbing for a more direct way to do this.

**Reply by foufrix** [reply]
Hi, I'm looking as well for a French model. How is it possible to help with that? I'm interested in sentiment analysis

**Reply by rachnachakraborty** [reply]
Hi,

Thanks for your nudge! Due to our day job pressures, initial plan got sidelined. We are are gearing up to initiate this journey right away. Can we collaborate on either email (wink@graype.in) or zulip chat? 

Best,
Rachna

**Reply by Fgabz** [reply]
Hello,

I'm interested in the progress for the french language! Any updates ?

**Reply by sanjayaksaxena** [reply]
@Fgabz We have initiated the work on first step — tokenization. Will keep you posted. What elements (e.g. sbd, ner, cer, pos, sentiment, word vectors) of pipeline are priority for you?

**Comment by boardmain** [comment]
hi, how collaborate to add italian model?

**Reply by sanjayaksaxena** [reply]
Request you to connect either on email or via Zulip.
